rmgpy/reaction.py

```diff
@@ -636,13 +636,31 @@ def get_equilibrium_constant(self, T, type='Kc', surface_site_density=2.5e-05):
         dN_surf = number_of_surface_products - number_of_surface_reactants # change in mols of surface spcs
         dN_gas = number_of_gas_products - number_of_gas_reactants # change in mols of gas spcs
 
+        # Determine the multiplication factor of the binding site^(-stoichiometric coefficient)
+        # (only relevant for reactions involving multidentate adsorbates)
+        unique_species = []
+        sigma_nu = 1
+        for product in self.products:
+            if len(product.molecule) != 0:
+                if product.contains_surface_site() and product not in unique_species:
+                    unique_species.append(product)
+                    sigma_nu *= product.get_binding_sites() ** (-self.get_stoichiometric_coefficient(product))
+
+        for reactant in self.reactants:
+            if len(reactant.molecule) != 0:
+                if reactant.contains_surface_site() and reactant not in unique_species:
+                    unique_species.append(reactant)
+                    sigma_nu *= reactant.get_binding_sites() ** (-self.get_stoichiometric_coefficient(reactant))
+
```

Comment:
I think this might add ineficciency and a bug.
If we have `A + A <=> B + A`.
we loop through each product (B then A)
if it's the first time we've seen it (not in unique_species) then we multiply by $\sigma ^ \nu$.
Works for B, then works for A.
then we loop the reactants (A then A).
First the bug - the unique_species list isn't reset, so it's skipped because we saw A as a product.
That's easily fixed: reset the list.
But then the inefficiency: 
Rather than just do the multiplication twice (once for each instance of A) we check if it's the first time we see A, then ask for get_stoichiometric_coefficient, which ..... aaaah. I take it back... the inefficiency cancels the bug, because of the way get_stoichiometric_coefficient returns the net value.
We already took care of all the A's.
OK. So I now think it's probably bug-free, but I think perhaps still a bit inefficient.
`get_stoichiometric_coefficient` loops through all the reactants and products every time it is called, counting how many times the requested species shows up.  Also, raising things to exponentials with `**` isn't fast. 

But correct is better than fast and wrong (probably, usually), so I don't mind you merging this and we speed it up later. I don't have time now.

And maybe what I think is faster, isn't much.

Reply:
I don't think this will be much slower than the alternative, and this is clear in what it does. I vote that this goes in as-is.